### PR TITLE
Feature/ELG-46/implement package calculator for combination heaters

### PR DESCRIPTION
**Pull request by FernandoRamirez-De-Aguilar**
Energy label calculator for space combination heaters in 2 flavours: Boilers & heat pumps
 
This is how the heat pump version looks, boiler is similar but with corresponding fields:
localhost_8080_categories_space-heaters_package-combination-heater_heat-pump_calculator

### CODE REVIEW

File: src/main/java/uk/gov/beis/els/categories/spaceheaters/model/BoilerCombinationCalculatorForm.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import javax.validation.constraints.NotNull;
+import uk.gov.beis.els.api.common.ApiValuesFromLoadProfileList;
+import uk.gov.beis.els.categories.waterheaters.service.WaterHeatersService;
+import uk.gov.beis.els.model.meta.FieldPrompt;
+
+public class BoilerCombinationCalculatorForm extends BoilerPackagesCalculatorForm {
+
+  @FieldPrompt("Water heating energy efficiency of water heater (%)")
+  @Digits(integer = 3, fraction = 0, message = "Enter the water heating efficiency (%), up to 3 digits long")
+  @Schema(type = "integer")
```
Comment (FernandoRamirez-De-Aguilar) [inline src/main/java/uk/gov/beis/els/categories/spaceheaters/model/BoilerCombinationCalculatorForm.java:15]: I left the schema values from the water heaters form just to make it quicker to make the new API endpoints

File: src/main/resources/templates/categories/space-heaters/boilerCombinationCalculator.ftl [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+<@common.standardProductForm
+title="Packages of combination heater, temperature control and solar device label calculator: ${preferentialHeater}"
+showInsetText=false
+isPackageCalculatorForm=true
+>
+    <@govukTextInput.textInput path="form.preferentialHeaterHeatOutput"/>
+    <@govukTextInput.textInput path="form.preferentialHeaterSeasonalSpaceHeatingEfficiencyPercentage"/>
+
+    <@govukFieldset.fieldset legendHeading="Solar Collector" legendSize="h2">
```
Comment (matteason) [inline src/main/resources/templates/categories/space-heaters/boilerCombinationCalculator.ftl:13]: Lower case c, `Solar collector`

File: src/main/resources/templates/categories/space-heaters/heatPumpCombinationCalculator.ftl [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+<#import 'heatPumpPackageAndCombinationCommon.ftl' as heatPumpCommon>
+
+<@common.standardProductForm
+title="Packages of combination heater, temperature control and solar device label calculator: ${preferentialHeater}"
+showInsetText=false
+isPackageCalculatorForm=true
+>
+    <@heatPumpCommon.commonHeatPumpPackageAndCombinationFields/>
+
+    <@govukFieldset.fieldset legendHeading="Solar Collector" legendSize="h2">
```
Comment (matteason) [inline src/main/resources/templates/categories/space-heaters/heatPumpCombinationCalculator.ftl:12]: Lower case c, `Solar collector`

File: src/main/resources/templates/categories/space-heaters/spaceHeaterPackagesCalculatorCommon.ftl [hunk 4]
```diff
@@ -1,15 +1,17 @@
...
     <@govukTextInput.textInput path="form.preferentialHeaterHeatOutput"/>
     <@govukTextInput.textInput path="form.preferentialHeaterSeasonalSpaceHeatingEfficiencyPercentage"/>
   </#if>
 
-  <@govukFieldset.fieldset legendHeading="Solar Collector" legendSize="h2">
-    <@govukTextInput.textInput path="form.solarCollectorSize"/>
-    <@govukTextInput.textInput path="form.solarCollectorEfficiencyPercentage"/>
-  </@govukFieldset.fieldset>
+  <#if showSolarCollector>
+    <@govukFieldset.fieldset legendHeading="Solar Collector" legendSize="h2">
```
Comment (matteason) [inline src/main/resources/templates/categories/space-heaters/spaceHeaterPackagesCalculatorCommon.ftl:10]: Lower case c, `Solar collector`

**Review by matteason — CHANGES_REQUESTED** [review]

**Review by matteason — APPROVED** [review]

**Pull request merged**